src/SleepForDays/SleepForDays.workflow.cs

```diff
@@ -0,0 +1,33 @@
+namespace TemporalioSamples.SleepForDays;
+
+using Microsoft.Extensions.Logging;
+using Temporalio.Workflows;
+
+[Workflow]
+public class SleepForDaysWorkflow
+{
+    private bool isComplete;
+
+    [WorkflowRun]
+    public async Task RunAsync()
+    {
+        isComplete = false;
+        while (!isComplete)
+        {
+            await Workflow.ExecuteActivityAsync(
+                (SleepForDaysActivities act) => act.SendEmail("Sleeping for 30 days"),
+                new() { StartToCloseTimeout = TimeSpan.FromSeconds(10) });
+            await Task.WhenAny(
+                Workflow.DelayAsync(TimeSpan.FromSeconds(30)),
+                Workflow.WaitConditionAsync(() => isComplete));
```

Comment:
There is an overload of `WaitConditionAsync` that accepts a timeout

Reply:
We considered this, but figured having a direct `sleep` more clearly showed the ability to wait for long periods of time.

Reply:
The concern here is that we usually encourage samples to look like a user would write them (otherwise we could have left timeout off of the wait condition calls). It also depends on if we're making a simple sample like for the website or not. If so and we want a clear "sleep" for this sample, we should consider using workflow cancellation as the interrupt mechanism because, even though we may not recommend cancellation as control flow, it makes the sample much more legible.

Reply:
Yeah that's true. The intention is for this to be a reference for the web sample and the discussion on Slack leaned towards using the workflow function and stripping out the signal stuff if it's too complicated.

But imo I agree. I err on the side of no interrupt mechanism at all, would be the simplest to read for a new viewer. But we have some consensus on this already and some samples already merged. I don't think it's worth re-opening the discussion or being inconsistent with the other samples, especially given the scope.

Reply:
Won't hold up this PR on this FWIW, but when we get to the public site content, I think it may not look like a sample

Reply:
Yeah I think that's fair. Marketing team asked for PRs yesterday which I shared, with a little addendum that if they want simpler examples I can remove the signal handling stuff from the workflow and send them code snippets